src/celllistimpl.h

```diff
@@ -789,14 +810,14 @@ template <class TBase> class CellListReverseMap : public TBase {
     void update(const Member& member, const CellIndex& new_cell_index) {
         const auto old_cell_index = member2cell.at(member);
         if (new_cell_index != old_cell_index) {
-            TBase::update(member, old_cell_index, new_cell_index);
+            TBase::move(member, old_cell_index, new_cell_index);
             member2cell.at(member) = new_cell_index;
         }
     }
 
   private:
     // TODO as a vector because size of Particle vector is constant
-    std::map<Member, CellIndex> member2cell; //!< mapping from a member to its cell
+    std::unordered_map<Member, CellIndex> member2cell; //!< mapping from a member to its cell
```

Comment:
Thought it would for large number of active particles because std::map has O(logN) complexity of insertions and deletions. But I have not found yet a system where this would appear, since it takes only very little time compared to the rest of calculation. I think I will try it with profiling in clion, but I have never used anything like that so it is going to take me a while.